clinica/iotools/converters/adni_to_bids/adni_modalities/adni_fmap.py

```diff
@@ -245,165 +253,235 @@ def fmap_image(
+        new_name = cut + one_phase_two_magnitudes_renamer(extension) + type
+        os.rename(fmap_path / previous_filename, fmap_path / new_name)
+
+
+def two_phases_two_magnitudes_renamer(old_extension: str) -> str:
+    if old_extension == "e1":
+        return "magnitude1"
+    elif old_extension == "e2":
+        return "magnitude2"
+    elif old_extension == "e1_ph":
+        return "phase1"
+    elif old_extension == "e2_ph":
+        return "phase2"
+
+
+def two_phases_two_magnitudes_handler(fmap_path: Path):
+    "Performs the checks and renaming for BIDS spec case 2 for fieldmaps"
+
+    sub = fmap_path.parent.parent.name
+    ses = fmap_path.parent.name
+    cprint(
+        f"BIDS Case 2 : expecting 2 phase and 2 magnitude files for {sub}, {ses}.",
+        lvl="info",
+    )
+
+    files = [f for f in os.listdir(fmap_path) if not f.startswith(".")]
+    check_json = [f for f in files if "ph.json" in f]
+    # Checking for Echotime keys in phase jsons
+    for js in check_json:
+        with open(fmap_path / js, "r") as file:
+            json_data = json.load(file)
+        if "EchoTime" not in json_data:
+            cprint(f'Invalid file {js}, missing "EchoTime" key.', lvl="warning")
+            unrecognized_fmap_case(fmap_path)
+            return
+    # Renaming
+    pattern = r"(.*_)fmap_(.*?)(\..*)"
+    for previous_filename in files:
+        rgx = re.search(pattern, previous_filename)
+        cut, extension, type = rgx.group(1), rgx.group(2), rgx.group(3)
+        new_name = cut + two_phases_two_magnitudes_renamer(extension) + type
+        os.rename(fmap_path / previous_filename, fmap_path / new_name)
+
+
+def direct_fieldmaps_renamer(old_extension: str) -> str:
+    if old_extension == "":
+        return "fieldmap"
+    else:
+        return "magnitude"
+
+
+def direct_fieldmaps_handler(fmap_path: Path):
+    sub = fmap_path.parent.parent.name
+    ses = fmap_path.parent.name
+
+    cprint(
+        f"BIDS Case 3: expecting 1 magnitude and 1 fieldmap files for {sub}, {ses}.",
+        lvl="info",
+    )
+
+    files = [f for f in os.listdir(fmap_path) if not f.startswith(".")]
+    # Assuming filename ends with "fmap" for the fieldmap
+    js = [f for f in files if f.endswith("fmap.json")][0]
+
+    # Checking for 'Units' key in fmap json
+    with open(fmap_path / js, "r") as file:
+        json_data = json.load(file)
+    if "Units" not in json_data:
+        cprint(
+            f'Invalid file {js} for Direct Fieldmapping, missing "Units" key.',
+            lvl="warning",
+        )
+        not_supported_handler(fmap_path)
+        return
+    # Renaming
+    pattern = r"(.*_)fmap(.*?)(\..*)"
+    for previous_filename in files:
+        rgx = re.search(pattern, previous_filename)
+        cut, extension, type = rgx.group(1), rgx.group(2), rgx.group(3)
+        new_name = cut + direct_fieldmaps_renamer(extension) + type
+        os.rename(fmap_path / previous_filename, fmap_path / new_name)
```

Comment:
I modified this a bit since I removed the renaming for case 3. We end up with only two cases that are very similar so I made a simpler renaming function handling both cases and did not use any factory.

WDYT ?